test/connectObservable.test.tsx

```diff
@@ -35,6 +35,35 @@ describe("connectObservable", () => {
     expect(result.current).toEqual(1)
   })
 
+  it("Only update when the previous and current update are distinct according to the comparator function", async () => {
+    interface TestUpdate {
+      value: number
+      valueToIgnore: string
+    }
+    const stream$ = new BehaviorSubject<TestUpdate>({
+      value: 0,
+      valueToIgnore: "A",
+    })
+
+    const compare = (a: TestUpdate, b: TestUpdate) => a.value === b.value
+    const [useLatestValue] = connectObservable(stream$, { compare })
+
+    const { result, rerender } = renderHook(() => useLatestValue())
+    expect(result.current.valueToIgnore).toEqual("A")
+    expect(result.current.value).toEqual(0)
+
+    stream$.next({ value: 0, valueToIgnore: "B" })
+    rerender()
```

Comment:
```suggestion
    act(() => {
      stream$.next({ value: 0, valueToIgnore: "B" })
    })
```

The "proper" way to do this stuff with `@testing-library/react-hooks` is to wrap the `stream@.next` on an `act` callback and not to call `rerender()`